packages/main/src/plugin/mcp/package/npm-spawner.ts

```diff
@@ -0,0 +1,72 @@
+/**********************************************************************
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+import { exec } from 'node:child_process';
+
+import { StdioClientTransport } from '@modelcontextprotocol/sdk/client/stdio.js';
+import type { Transport } from '@modelcontextprotocol/sdk/shared/transport.js';
+
+import { MCPSpawner } from './mcp-spawner.js';
+
+const NPX_COMMAND = 'npx';
+
+export class NPMSpawner extends MCPSpawner<'npm'> {
+  #disposables: Array<AsyncDisposable> = [];
+
+  async spawn(): Promise<Transport> {
+    if (!this.pack.identifier) throw new Error('missing identifier in MCP Local Server configuration');
+    if (this.pack.fileSha256) {
+      console.warn('specified file sha256 is not supported with npx spawner');
+    }
+
+    const transport = new StdioClientTransport({
+      command: NPX_COMMAND,
+      args: [
+        ...(this.pack.runtimeArguments ?? []),
+        // let's use package@version if version is specified
+        this.pack.version ? `${this.pack.identifier}@${this.pack.version}` : this.pack.identifier,
+        ...(this.pack.packageArguments ?? []),
+      ],
+      env: this.pack.environmentVariables,
+    });
+    this.#disposables.push({
+      [Symbol.asyncDispose]: () => {
+        return transport.close();
+      },
+    });
+    return transport;
+  }
+
+  async enabled(): Promise<boolean> {
+    const { promise, resolve } = Promise.withResolvers<boolean>();
+
+    // eslint-disable-next-line sonarjs/os-command
+    exec(`${NPX_COMMAND} --version`, error => {
```

Comment:
issue (blocking): The npx command is expensive and it seems that the enabled method is called for every `packages/main/src/plugin/mcp/mcp-registry.ts:219` server (could be thousands) of the registry

Reply:
It is only called for each server configured, so not for all server in the registry